D1371R0.pandoc

```diff
@@ -1082,6 +1082,57 @@ it is to `vector<T>` as far as the type system is concerned. This limitation
 will likely be even bigger in a pattern matching context with the amount of
 customization points available for user-defined behavior.
 
+## Unrestricted Side Effects
+
+We considered the possibility of restricting side-effects within patterns.
+Specifically, we discussed whether modifying the value currently being matched
+in the middle of evaluation should have defined behavior.
+
+The consideration was due to potential optimization opportunities.
+
+Consider the following:
+
+```cpp
+int x = 1;
+
+inspect (x) {
+    0: std::cout << 0;
+    1 if (test()): std::cout << 1;
+    2: std::cout << 2;
+}
+```
+
+If modifying the value currently being matched has undefined behavior,
+a compiler can assume that `test` will not change the value of `x`.
+This means that the above snippet could be translated to:
+
+```cpp
+int x = 1;
+
+switch (x) {
+    case 0: std::cout << 0; break;
+    case 1: if (test()) { std::cout << 1; } break;
+    case 2: std::cout << 2; break;
+}
+```
```

Comment:
I added the description, and added the following comment to the code snippet: `// A possible representation in C++` in order to not imply that compilers generate C++.

I've left the C++ representation because (at least to me) they demonstrate the contrast on sight.